### ISSUE TITLE
client:build does not work anymore

### ISSUE THREAD

**Issue by mipastgt** [gluonhq/client-maven-plugin#212]
If I follow the documentation and start the build with `mvn client:build` I immediately get the error message `release version 11.0.7 not supported` which is the current version of the VM used. In my case the latest GraalVM snapshot. If I use another VM the error is the same, only the number changes according to the VM version.

In order to get something built (which currently later fails for other reasons https://github.com/gluonhq/client-maven-plugin/issues/211) I have to explicitly split up the commands into `mvn client:compile` and  `mvn client:link`.

```
mpaus@ip11:~/Projects/mpMediaSoft/apps/mpCoPilot/mpCoPilot-main-mobile$ mvn client:build
[INFO] Scanning for projects...
[INFO] 
[INFO] --------< de.mpmediasoft:de.mpmediasoft.mpcopilot.main.mobile >---------
[INFO] Building mpCoPilotApp2 0.0.1-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- client-maven-plugin:0.1.29-SNAPSHOT:build (default-cli) @ de.mpmediasoft.mpcopilot.main.mobile ---
[INFO] Scanning for projects...
[INFO] 
[INFO] --------< de.mpmediasoft:de.mpmediasoft.mpcopilot.main.mobile >---------
[INFO] Building mpCoPilotApp2 0.0.1-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] >>> client-maven-plugin:0.1.29-SNAPSHOT:compile (default-cli) > process-classes @ de.mpmediasoft.mpcopilot.main.mobile >>>
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ de.mpmediasoft.mpcopilot.main.mobile ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 17 resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.1:compile (default-compile) @ de.mpmediasoft.mpcopilot.main.mobile ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 25 source files to /home/mpaus/Projects/mpMediaSoft/apps/mpCoPilot/mpCoPilot-main-mobile/target/classes
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.671 s
[INFO] Finished at: 2020-07-20T12:49:16+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project de.mpmediasoft.mpcopilot.main.mobile: Fatal error compiling: error: release version 11.0.7 not supported -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.948 s
[INFO] Finished at: 2020-07-20T12:49:16+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.gluonhq:client-maven-plugin:0.1.29-SNAPSHOT:build (default-cli) on project de.mpmediasoft.mpcopilot.main.mobile: Error, client:build failed -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
mpaus@ip11:~/Projects/mpMediaSoft/apps/mpCoPilot/mpCoPilot-main-mobile$ 
```

**Comment by jperedadnr** [comment]
Hmm, possible related to the latest PR merged, with this [commit](https://github.com/gluonhq/client-maven-plugin/pull/207/commits/c4c3fa2fcc741ac145c7bbf5713d80ddff403162).

Can you describe in more detail your issue? OS, exceptions, and how it can be reproduced? Also what it works for you.

**Comment by mipastgt** [comment]
This was done on Ubuntu 20.04 in order to try a build for Android. I'll try whether I can reproduce that with one of the Hello??? examples. What do you mean in your last sentence?

**Comment by jperedadnr** [comment]
If I get it right you said `mvn client:build` doesn't work, but `mvn client:compile client:link` does?

**Comment by mipastgt** [comment]
Yes, exactly.

**Comment by mipastgt** [comment]
I just verified that I get the same behaviour when I build for the host (Ubuntu) and not only when I build for Android from Linux. But I cannot reproduce this behaviour with a small example although I even ran that in the same Shell (so environment variables should be the same).

**Comment by jperedadnr** [comment]
As I mentioned above, there is this [commit](https://github.com/gluonhq/client-maven-plugin/pull/207/commits/c4c3fa2fcc741ac145c7bbf5713d80ddff403162) that adds the maven system properties to the invocationRequest (so client:build calls client:compile and client:link with the same properties for both goals).
In case your complex scenario uses some system properties that might cause some issues.
If you can run `mvn -X client:build` you will get more indications.

Based on the error from the compile plugin `release version 11.0.7 not supported`, looks like it is mixing some properties (should be 11 not 11.0.7)

**Comment by jperedadnr** [comment]
Should be fixed with #222
